operator/builtin/parser/xml/xml_test.go

```diff
@@ -0,0 +1,141 @@
+package xml
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParse(t *testing.T) {
+	testCases := []struct {
+		name           string
+		value          interface{}
+		expectedResult interface{}
+		expectedErr    error
+	}{
+		{
+			name:        "Empty string",
+			value:       "",
+			expectedErr: errors.New("failed to decode as xml"),
+		},
+		{
+			name:        "No nodes",
+			value:       "regular string",
+			expectedErr: errors.New("no xml nodes found"),
+		},
+		{
+			name:        "Non string value",
+			value:       5,
+			expectedErr: errors.New("value is not a string"),
+		},
+		{
+			name:        "Incomplete node",
+			value:       "<person age='30'>",
+			expectedErr: errors.New("failed to get next xml token"),
+		},
+		{
+			name:  "Single node",
+			value: "<person age='30'>Jon Smith</person>",
+			expectedResult: map[string]interface{}{
+				"type": "person",
+				"attributes": map[string]string{
+					"age": "30",
+				},
+				"value": "Jon Smith",
+			},
+		},
+		{
+			name:  "Multiple nodes",
+			value: "<person age='30'>Jon Smith</person><person age='28'>Sally Smith</person>",
+			expectedResult: []map[string]interface{}{
+				{
+					"type": "person",
+					"attributes": map[string]string{
+						"age": "30",
+					},
+					"value": "Jon Smith",
+				},
+				{
+					"type": "person",
+					"attributes": map[string]string{
+						"age": "28",
+					},
+					"value": "Sally Smith",
+				},
+			},
+		},
+		{
+			name:  "Children nodes",
+			value: "<worker><person age='30'>Jon Smith</person></worker>",
+			expectedResult: map[string]interface{}{
+				"type": "worker",
+				"children": []map[string]interface{}{
+					{
+						"type": "person",
+						"attributes": map[string]string{
+							"age": "30",
+						},
+						"value": "Jon Smith",
+					},
+				},
+			},
+		},
+	}
```

Comment:
We should have a test case with multiple levels of children, to prove that the parse function will recurse.

Reply:
I updated the test for Mitch to include a second layer of embedding.